zip-0316.rst

```diff
@@ -416,8 +464,50 @@ Requirements for both Unified Addresses and Unified Viewing Keys
   that cannot be interpreted as specified above.
 
 
-Address hardening
------------------
+Adding new types
+----------------
+
+It is intended that new Receiver Types and Viewing Key Types SHOULD
+be introduced either by a modification to this ZIP or by a new ZIP,
+in accordance with the ZIP Process [#zip-0000]_.
+
+Experimental types MAY be added using the reserved Typecodes
+:math:`\mathtt{0xFFFA}` to :math:`\mathtt{0xFFFF}` inclusive. This
+provides for six simultaneous experiments, which can be referred to
+as experiments A to F. This should be sufficient because experiments
+are expected to be reasonably short-term, and should otherwise be
+either standardized in a ZIP (and allocated a Typecode outside this
+reserved range) or discontinued.
+
+
+Deriving a UIVK from a UFVK
+---------------------------
+
+Each UIVK Encoding can straightforwardly be derived from the
+corresponding UFVK Encoding, without changing the Type Code. In the
+case of a Transparent P2PKH UFVK Encoding, the UIVK Encoding is the
+same.
+
+
+Deriving a Unified Address from a UIVK
```

Comment:
TODO(@daira ) : check if FF1 is a S-PRP, want to avoid a possibly linkability attack if keys diversified in serial

Reply:
As far as I know this is not a significant concern. The best attacks on FF1 (e.g. up to 7 rounds out of 10 in https://link.springer.com/chapter/10.1007/978-3-319-63715-0_23) are only practical for much smaller domain sizes than 2<sup>88</sup>, so in practice there is a large safety margin that should be sufficient to protect against chosen ciphertext attacks.